sdk/purview/azure-purview-sharing/samples/received_shares_examples.py

```diff
@@ -0,0 +1,121 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+# These examples are ingested by the documentation system, and are
+# displayed in the SDK reference documentation. When editing these
+# example snippets, take into consideration how this might affect
+# the readability and usability of the reference documentation.
+
+# Create a received share client
+
+# [START create_a_received_share_client]
+import os
+
+from azure.purview.sharing import PurviewSharing
+from azure.identity import ClientSecretCredential
+
+endpoint = os.environ["ENDPOINT"]
+tenant_id = os.environ.get("AZURE_TENANT_ID", getattr(os.environ, "TENANT_ID", None))
+client_id = os.environ.get("AZURE_CLIENT_ID", getattr(os.environ, "CLIENT_ID", None))
+secret = os.environ.get("AZURE_CLIENT_SECRET", getattr(os.environ, "CLIENT_SECRET", None))
+credential = ClientSecretCredential(tenant_id=str(tenant_id), client_id=str(client_id), client_secret=str(secret))
+
+client = PurviewSharing(endpoint=endpoint,credential=credential)
+# [END create_a_received_share_client]
+
+# Get all detached received shares
+# [START get_all_detached_received_shares]
+from azure.purview.sharing.operations._operations import (
+    build_received_shares_list_detached_request
+)
+
+list_detached_request = build_received_shares_list_detached_request(orderby="properties/createdAt desc")
+list_detached_response = client.send_request(list_detached_request)
+# [END get_all_detached_received_shares]
+
+# Attach a received share
+# [START attach_a_received_share]
+import json
+
+from azure.core.exceptions import HttpResponseError
+from azure.purview.sharing.operations._operations import (
+    build_received_shares_create_or_replace_request
+)
+
+consumer_storage_account_resource_id = "/subscriptions/{subscription-id}/resourceGroups/consumer-storage-rg/providers/Microsoft.Storage/storageAccounts/consumerstorage"
+list_detached = json.loads(list_detached_response.content)
+received_share = list_detached['value'][0]
+
+store_reference = {
+    "referenceName": consumer_storage_account_resource_id,
+    "type": "ArmResourceReference"
+}
+
+sink = {
+    "properties": {
+        "containerName": "container-test",
+        "folder": "folder-test",
+        "mountPath": "mountPath-test",
+    },
+    "storeKind": "AdlsGen2Account",
+    "storeReference": store_reference
+}
+
+received_share['properties']['sink'] = sink
+
+update_request = build_received_shares_create_or_replace_request(
```

Comment:
you shouldn't need to use the request builders at all. Just directly use the operations on the client. This should be more like `client.received_shares.begin_create_or_replace` or whatever the operation is

Reply:
in general, don't expose private methods / imports (anything with "_" in the name) in code for users